src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java

```diff
@@ -574,14 +578,51 @@ default boolean addDestroyEffects(IBlockState state, World world, BlockPos pos,
     *   Plains check if its grass or dirt
     *   Water check if its still water
     *
-    * @param state The Current state
+    * @param state The state of this block, not to be confused with the state of the plant.
     * @param world The current world
-    *
     * @param facing The direction relative to the given position the plant wants to be, typically its UP
-    * @param plantable The plant that wants to check
+    * @param plantState The state of the plant, retrieved from an IPlantable (or the world if this plant was already placed).
+    * @param plant The plant that is attempting to be planted or stay planted.
     * @return True to allow the plant to be planted/stay.
     */
-    boolean canSustainPlant(IBlockState state, IBlockReader world, BlockPos pos, EnumFacing facing, IPlantable plantable);
+    default boolean canSustainPlant(IBlockState state, IBlockReader world, BlockPos pos, EnumFacing facing, IBlockState plantState, IPlant plant)
```

Comment:
Ya, literally all you do is remove the getBlockState call, and 1/2 implement in the patch....

Reply:
See comments on Block/BlockBush reviews